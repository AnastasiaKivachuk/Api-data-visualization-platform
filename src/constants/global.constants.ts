export const DATE_FORMAT = {
    DATE_TIME_NUMBER: 'DD.MM.YYYY, HH:mm A',
    DATE_WORD: 'MMM DD, YYYY',
};