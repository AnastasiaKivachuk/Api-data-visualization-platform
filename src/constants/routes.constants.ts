export const HOME_ROUTE = '/';

export const USERS_ROUTE = '/users';
export const USER_ROUTE = '/users/[id]';

export const LAUNCHES_ROUTE = '/launches';
export const LAUNCH_ROUTE = '/launches/[id]';