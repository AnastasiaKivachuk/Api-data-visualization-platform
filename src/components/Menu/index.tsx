import React from "react";
import Link from "next/link";
import styled from "@emotion/styled";

import Grid from '@mui/material/Grid'
import MuiAppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import Typography from '@mui/material/Typography'
import {IconButton} from "@mui/material";

import {APP_BAR_ITEMS} from "../../constants/menu.constants";
import {HOME_ROUTE} from "../../constants/routes.constants";

const Menu = (): JSX.Element => {

    const StyledWrapper = styled.div`
      display: flex;
      gap: 25px;
      width: fit-content;
    `

    const StyledWrapperLink = styled.div`
      align-content: center;
      cursor: pointer;
      display: flex;
    `

    return (
        <MuiAppBar>
            <Toolbar>
                <Grid container justifyContent="space-between" alignItems="center">
                    <StyledWrapperLink>
                    <Link href={HOME_ROUTE} color="inherit">
                        <a>
                    <Typography variant="h4">Space X</Typography>
                        </a>
                        </Link>
                    </StyledWrapperLink>
                    <StyledWrapper>
                        {APP_BAR_ITEMS.map(({link, text, icon}) =>
                            <Link key={link} href={link} color="inherit">
                               <a>
                                <StyledWrapperLink>
                                    <IconButton size="large">
                                        {icon}
                                    </IconButton>
                                    {text && <Typography variant="h6" lineHeight='48px'>{text}</Typography>}
                                </StyledWrapperLink>
                               </a>
                            </Link>)}
                    </StyledWrapper>
                </Grid>
            </Toolbar>
        </MuiAppBar>
    )
}

export default Menu
