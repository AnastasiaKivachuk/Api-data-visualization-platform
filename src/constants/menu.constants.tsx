import {HOME_ROUTE, LAUNCHES_ROUTE, USERS_ROUTE} from './routes.constants';
import GroupIcon from '@mui/icons-material/Group';
import HomeIcon from '@mui/icons-material/Home';
import RocketLaunchIcon from '@mui/icons-material/RocketLaunch';
import React from "react";

export const APP_BAR_ITEMS = [
    {
        link: HOME_ROUTE,
        text: '',
        icon: <HomeIcon/>
    },
    {
        link: USERS_ROUTE,
        text: 'Users',
        icon: <GroupIcon/>
    },
    {
        text: 'Launches',
        link: LAUNCHES_ROUTE,
        icon: <RocketLaunchIcon/>
    }
]