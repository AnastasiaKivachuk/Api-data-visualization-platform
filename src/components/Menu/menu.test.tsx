import {shallow} from "enzyme";

import {HOME_ROUTE, LAUNCHES_ROUTE, USERS_ROUTE} from "../../constants/routes.constants";
import Menu from "./index";


const routes = [HOME_ROUTE, HOME_ROUTE, USERS_ROUTE, LAUNCHES_ROUTE]

const component = shallow(<Menu/>);

describe("Menu component", () => {
        it("should render Menu component", () => {
            expect(component).toMatchSnapshot();
        })
        it("check all elements", () => {
            expect(component.find('a').length).toBe(4)
            component.find('ForwardRef').map((item, index) =>
                expect(item.prop('href')).toBe(routes[index]))
        })
    }
)