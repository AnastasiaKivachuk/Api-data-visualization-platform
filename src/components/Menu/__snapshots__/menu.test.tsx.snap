// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should render Menu component 1`] = `
<ForwardRef(AppBar)>
  <ForwardRef(Toolbar)>
    <ForwardRef(Grid)
      alignItems="center"
      container={true}
      justifyContent="space-between"
    >
      <Styled(div)>
        <ForwardRef
          color="inherit"
          href="/"
        >
          <a>
            <ForwardRef(Typography)
              variant="h4"
            >
              Space X
            </ForwardRef(Typography)>
          </a>
        </ForwardRef>
      </Styled(div)>
      <Styled(div)>
        <ForwardRef
          color="inherit"
          href="/"
          key="/"
        >
          <a>
            <Styled(div)>
              <ForwardRef(IconButton)
                size="large"
              >
                <Memo(ForwardRef(HomeIcon)) />
              </ForwardRef(IconButton)>
            </Styled(div)>
          </a>
        </ForwardRef>
        <ForwardRef
          color="inherit"
          href="/users"
          key="/users"
        >
          <a>
            <Styled(div)>
              <ForwardRef(IconButton)
                size="large"
              >
                <Memo(ForwardRef(GroupIcon)) />
              </ForwardRef(IconButton)>
              <ForwardRef(Typography)
                lineHeight="48px"
                variant="h6"
              >
                Users
              </ForwardRef(Typography)>
            </Styled(div)>
          </a>
        </ForwardRef>
        <ForwardRef
          color="inherit"
          href="/launches"
          key="/launches"
        >
          <a>
            <Styled(div)>
              <ForwardRef(IconButton)
                size="large"
              >
                <Memo(ForwardRef(RocketLaunchIcon)) />
              </ForwardRef(IconButton)>
              <ForwardRef(Typography)
                lineHeight="48px"
                variant="h6"
              >
                Launches
              </ForwardRef(Typography)>
            </Styled(div)>
          </a>
        </ForwardRef>
      </Styled(div)>
    </ForwardRef(Grid)>
  </ForwardRef(Toolbar)>
</ForwardRef(AppBar)>
`;
